core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -469,14 +469,18 @@ class ReplicaManager(val config: KafkaConfig,
     })
   }
 
-  private def completeDelayedOperationsWhenNotPartitionLeader(topicPartition: TopicPartition, topicId: Option[Uuid]): Unit = {
+  private def completeDelayedOperationsWhenNotPartitionLeader(topicPartition: TopicPartition, topicId: Option[Uuid],
+                                                              isStopReplica: Boolean): Unit = {
     val topicPartitionOperationKey = TopicPartitionOperationKey(topicPartition)
     delayedProducePurgatory.checkAndComplete(topicPartitionOperationKey)
-    delayedFetchPurgatory.checkAndComplete(topicPartitionOperationKey)
-    delayedRemoteFetchPurgatory.checkAndComplete(topicPartitionOperationKey)
-    delayedRemoteListOffsetsPurgatory.checkAndComplete(topicPartitionOperationKey)
-    if (topicId.isDefined) delayedShareFetchPurgatory.checkAndComplete(
-      new DelayedShareFetchPartitionKey(topicId.get, topicPartition.partition()))
+    // We support fetch from follower now, So only 'stop reply' requires the following things
+    if (isStopReplica) {
+      delayedFetchPurgatory.checkAndComplete(topicPartitionOperationKey)
```

Comment:
Pardon me, I have a question. Some tests case are checking that the produce failed because it changed to follower before replicating, if we won't completion this fetch purgatory, and we can't observe the error about `NOT_LEADER_OR_FOLLOWER`.  WDYT?

Reply:
@m1a2st  Thanks very much for code review, I will check the failed test cases and make some modifications.